fe/fe-core/src/main/java/com/starrocks/statistic/HistogramStatisticsCollectJob.java

```diff
@@ -0,0 +1,73 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.statistic;
+
+import com.starrocks.analysis.StatementBase;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.cluster.ClusterNamespace;
+import com.starrocks.common.DdlException;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.qe.QueryState;
+import com.starrocks.qe.StmtExecutor;
+import org.apache.velocity.VelocityContext;
+
+import java.util.List;
+
+import static com.starrocks.sql.parser.SqlParser.parseFirstStatement;
+import static com.starrocks.statistic.Constants.HistogramStatisticsTableName;
+
+public class HistogramStatisticsCollectJob extends StatisticsCollectJob {
+    private static final String COLLECT_HISTOGRAM_STATISTIC_TEMPLATE =
+            "SELECT $tableId, '$columnName', '$dbName.$tableName', histogram($columnName, $totalRows, $sampleRows, $bucketNum)"
+                    + " FROM (SELECT * FROM $dbName.$tableName $sampleTableHint ORDER BY $columnName LIMIT $totalRows) t";
+
+    public HistogramStatisticsCollectJob(AnalyzeJob analyzeJob, Database db, OlapTable table, List<String> columns) {
+        super(analyzeJob, db, table, columns);
+    }
+
+    @Override
+    public void collect() throws Exception {
+        ConnectContext context = StatisticUtils.buildConnectContext();
+        context.getSessionVariable().setNewPlanerAggStage(1);
+
+        long totalRows = table.getRowCount();
+        long sampleRows = Long.parseLong(analyzeJob.getProperties().get(Constants.PROP_SAMPLE_COLLECT_ROWS_KEY));
+        long bucketNum = Long.parseLong(analyzeJob.getProperties().get(Constants.PRO_BUCKET_NUM));
+
+        for (String column : analyzeJob.getColumns()) {
+            String sql = buildCollectHistogram(db, table, totalRows, sampleRows, bucketNum, column);
```

Comment:
we can not collect full histogram?

Reply:
There is a global sorting process in the histogram calculation, so only one column can be used as the sorting column at a time